lua/snacks/gitbrowse.lua

```diff
@@ -70,12 +70,17 @@ function M.open(opts)
     if name and remote then
       local url = M.get_url(remote, opts)
       if url:find("github") and branch and branch ~= "master" and branch ~= "main" then
-        url = ("%s/tree/%s"):format(url, branch)
+        url = ("%s/tree/%s/"):format(url, branch)
+      elseif branch == "main" then
+        url = ("%s/tree/main/"):format(url)
+      elseif branch == "master" then
+        url = ("%s/tree/master/"):format(url)
```

Comment:
Why do main and master need a special treatment?

Reply:
The url is different?

Reply:
He means that this:

```lua
 if url:find("github") and branch and branch ~= "master" and branch ~= "main" then
        url = ("%s/tree/%s/"):format(url, branch)
      elseif branch == "main" then
        url = ("%s/tree/main/"):format(url)
      elseif branch == "master" then
        url = ("%s/tree/master/"):format(url)
```

Can be replaced by this:

```lua
 if url:find("github") and branch then
        url = ("%s/tree/%s/"):format(url, branch)
```

Either way, this stuff currently only works for Github. I'll se if I can make it a bit more general to work with others as well.

Reply:
Oh I see. Thanks for clarifying the logic. Much appreciate it.

Reply:
@folke Should I close this maybe, since you're going to come up with a better implementation? Or just close it yourself. I really don't mind.